app/webpacker/components/Panel/SeniorDelegate/Regions/index.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect } from 'react';
+
+import { Dropdown, Header } from 'semantic-ui-react';
+
+import useInputState from '../../../../lib/hooks/useInputState';
+import useLoadedData from '../../../../lib/hooks/useLoadedData';
+import { apiV0Urls } from '../../../../lib/requests/routes.js.erb';
+import { delegateRegionsStatus, groupTypes } from '../../../../lib/wca-data.js.erb';
+import Errored from '../../../Requests/Errored';
+import Loading from '../../../Requests/Loading';
+import Region from './Region';
+
+export default function Regions({ loggedInUserId }) {
+  const {
+    data: seniorDelegateRoles,
+    loading, error,
+  } = useLoadedData(apiV0Urls.userRoles.listOfUser(
+    loggedInUserId,
+    'groupName', // Sort params
+    {
+      isActive: true,
+      isGroupHidden: false,
+      status: delegateRegionsStatus.senior_delegate,
+      groupType: groupTypes.delegate_region,
+    },
+  ));
+  const [selectedGroupIndex, setSelectedGroupIndex] = useInputState();
+
+  useEffect(() => {
+    if (seniorDelegateRoles?.length > 0) setSelectedGroupIndex(0);
+  }, [seniorDelegateRoles, setSelectedGroupIndex]);
+
+  if (!loading && seniorDelegateRoles.length === 0) return <p>You cannot manage any regions..</p>;
+  if (loading || selectedGroupIndex === undefined) return <Loading />;
+  if (error) return <Errored />;
+
+  return (
+    <>
+      <Header>Regions</Header>
+      <div>
+        <Dropdown
+          options={seniorDelegateRoles.map((role, index) => ({
+            key: role.id,
+            text: role.group.name,
+            value: index,
+          }))}
```

Comment:
Move this up into a `useMemo`?

Reply:
I don't think we need to memoize things as much as we've been doing lately (but I also think it doesn't really hurt to do it). React is designed to handle re-renders efficiently, and our components and arrays are pretty small.
(Of course, when I try to find a reference to link to, I find tons of people arguing each side, so who knows for sure.)